executor/mpp_gather.go

```diff
@@ -159,7 +166,14 @@ func (e *MPPGather) Open(ctx context.Context) (err error) {
 // Next fills data into the chunk passed by its caller.
 func (e *MPPGather) Next(ctx context.Context, chk *chunk.Chunk) error {
 	err := e.respIter.Next(ctx, chk)
-	return errors.Trace(err)
+	if err != nil {
+		return err
+	}
+	err = table.FillVirtualColumnValue(e.virtualColumnRetFieldTypes, e.virtualColumnIndex, e.schema.Columns, e.columns, e.ctx, chk)
```

Comment:
I'm not sure if this `FillVirtualColumnValue` can work correctly if the mpp request is complex, for example assuming that virutal_col1 is defined as a + b in table t2, will the current pr support `select t1.a, t1.b, t2.virual_col1, t2.a, t2.b from t1 join t2 where xxx`?